sdk/core/azure-core-amqp/inc/azure/core/amqp/link.hpp

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-Licence-Identifier: MIT
+
+#pragma once
+
+#include "models/amqp_value.hpp"
+
+#include <chrono>
+#include <memory>
+#include <string>
+#include <vector>
+struct LINK_INSTANCE_TAG;
+namespace Azure { namespace Core { namespace Amqp {
+  namespace _internal {
+    class Session;
+    struct LinkEndpoint;
+    enum class SenderSettleMode;
+    enum class ReceiverSettleMode;
+
+    enum class LinkDurability
+    {
+      None,
+      Configuration,
+      UnsettledState
+    };
+
+    enum class SessionRole
+    {
+      Sender,
+      Receiver
+    };
+  } // namespace _internal
+
+  namespace _detail {
+    class LinkImpl;
+
+    enum class LinkState
+    {
+      Invalid,
+      Detached,
+      HalfAttachedAttachSent,
+      HalfAttachAttachReceived,
+      Attached,
+      Error,
+    };
+    enum class LinkTransferResult
+    {
+      Error,
+      Busy
+    };
+
+    enum class LinkDeliverySettleReason
+    {
+      DispositionReceived,
+      Settled,
+      NotDelivered,
+      Timeout,
+      Cancelled,
+    };
```

Comment:
What does it mean for something to "settle", across a link?

Reply:
I wish that explained what it means for something to be settled, but it only highlights the modes. This is the type of documentation that I find useless, and a bit annoying:
```text
unsettled - sender sends it initially unsettled
settled - sender sends it settled
mixed - mix of settled and unsettled
```

That is not helpful in understanding the concept at all as it is essentially restating the obvious. We should also **try** to avoid doing that in our doc comments (like url param - the Url parameter).